### PR TITLE
MMP-501: Support testJar to include test dependencies into heavyweigh…

### PR DESCRIPTION
**Pull request by gsfernandes**
…t and also build classloader model json when useLocationRepository and lightweight is enabled

### CODE REVIEW

File: mule-packager/src/main/java/org/mule/tools/api/repository/RepositoryGenerator.java [hunk 1]
```diff
@@ -58,15 +58,17 @@ public RepositoryGenerator(File projectPomFile, File outputDirectory, ArtifactIn
...
     ApplicationClassloaderModel appModel =
-        applicationClassLoaderModelAssembler.getApplicationClassLoaderModel(projectPomFile, outputDirectory, appGAVModel);
+        applicationClassLoaderModelAssembler.getApplicationClassLoaderModel(projectPomFile, outputDirectory, appGAVModel, false);
     installArtifacts(getRepositoryFolder(), artifactInstaller, appModel, false);
     return appModel.getClassLoaderModel();
   }
 
-  public ClassLoaderModel generate(boolean lightweight, boolean useLocalRepository, boolean prettyPrinting)
+  public ClassLoaderModel generate(boolean lightweight, boolean useLocalRepository, boolean prettyPrinting,
+                                   boolean includeTestScopeDependencies)
```
Comment (diegostrubolini) [inline mule-packager/src/main/java/org/mule/tools/api/repository/RepositoryGenerator.java]: just for consistency I would rename this to includeTestDependencies

File: mule-artifact-it/mule-packager-it/src/test/resources/expected-test-jar-classloader-model-project/META-INF/mule-src/test-jar-classloader-model-project/log.txt [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+[INFO] ------------------------------------------------------------------------
+[INFO] 
+[INFO] --- maven-clean-plugin:2.5:clean (default-cli) @ empty-classloader-model-project ---
+[INFO] Deleting /Users/ms0374/Documents/projects/mule-artifact-tools/mule-artifact-it/target/test-classes/empty-classloader-model-project/target
+[INFO] 
+[INFO] --- mule-maven-plugin:3.0.0-SNAPSHOT:validate (default-validate) @ empty-classloader-model-project ---
+[INFO] 
+[INFO] --- mule-maven-plugin:3.0.0-SNAPSHOT:initialize (default-initialize) @ empty-classloader-model-project ---
+[INFO] 
+[INFO] --- mule-maven-plugin:3.0.0-SNAPSHOT:generate-sources (default-generate-sources) @ empty-classloader-model-project ---
```
Comment (lucianoRM) [inline mule-artifact-it/mule-packager-it/src/test/resources/expected-test-jar-classloader-model-project/META-INF/mule-src/test-jar-classloader-model-project/log.txt]: Is this file necessary?

File: mule-artifact-it/mule-packager-it/src/test/resources/expected-test-jar-lightweight-local-repository-classloader-model-project/META-INF/mule-src/test-jar-classloader-model-project/log.txt [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+[INFO] ------------------------------------------------------------------------
+[INFO] 
+[INFO] --- maven-clean-plugin:2.5:clean (default-cli) @ empty-classloader-model-project ---
+[INFO] Deleting /Users/ms0374/Documents/projects/mule-artifact-tools/mule-artifact-it/target/test-classes/empty-classloader-model-project/target
+[INFO] 
+[INFO] --- mule-maven-plugin:3.0.0-SNAPSHOT:validate (default-validate) @ empty-classloader-model-project ---
+[INFO] 
+[INFO] --- mule-maven-plugin:3.0.0-SNAPSHOT:initialize (default-initialize) @ empty-classloader-model-project ---
+[INFO] 
+[INFO] --- mule-maven-plugin:3.0.0-SNAPSHOT:generate-sources (default-generate-sources) @ empty-classloader-model-project ---
```
Comment (lucianoRM) [inline mule-artifact-it/mule-packager-it/src/test/resources/expected-test-jar-lightweight-local-repository-classloader-model-project/META-INF/mule-src/test-jar-classloader-model-project/log.txt]: this too
